TechnicalDocuments/UI_engineering_guidelines.md

```diff
@@ -0,0 +1,37 @@
+# UI Engineering Guidelines
+
+#### Updated at: 3-04-2019
+
+## The purpose of this document
+The purpose of this document is to describe how we implement our UI in babylon and which technologies we should use.
+
+## Nomenclature
+
+### BentoKit
+
+`BentoKit` is the first implementation of `Bento` Components in Babylon.
+Its purpose is to provide __convenient__ components in order to assist developers.
+`BentoKit` isn't suitable for Component composition. 
+
+### Atomic Components
+`Atomic Components` are the newest implementation of `Bento` Components in Babylon.
+`Atomic Components` are lightweight components, they are the `Bento` equivalent of standard UIKit components. Also they can be used in `UIStackView`-based Component Composition.
```

Comment:
Not directly related to this document but imo it's a bit confusing when there is `AtomicComponents.swift` which only contains `UI` namespace, but there are bunch of other components like `Width`, `Align`, `Stack` mixed with BentoKit-like components in BabylonUI. Shouldn't they be considered as atomic as well? I'd suggest rather than to group all of them in a folder `AtomicComponents` to make the distinction clear. They all should be probably moved under "DesignLibrary" group as well. Also for example `Stack` component can be moved under `UI` namespace as it is just a wrapper around `UIStackView`. cc @sergdort

Reply:
> They all should be probably moved under "DesignLibrary" group as well

Why?

`DesignLibrary` is just an abstraction. It could also use `BentoKit`.

>Not directly related to this document but imo it's a bit confusing when there is AtomicComponents.swift which only contains UI namespace, but there are bunch of other components like Width, Align, Stack mixed with BentoKit-like components in BabylonUI. Shouldn't they be considered as atomic as well? 

Yes they are related but `Width` and `Stack` can also be used with `AnyRenderable`.
If the component is using `HeightCustomizing` then yes in some cases in may not work.

Reply:
Because they all are "atomic" in the sense that they are providing one specific and very small functionality. And they are all part of design library. I can make a PR with how I see the structure. It's only about files grouping though, but will improve discoverability imo.